src/index/hnsw.h

```diff
@@ -276,7 +276,7 @@ class HNSW : public Index {
     bool is_init_memory_ = false;
     DataTypes type_;
 
-    std::shared_ptr<SafeAllocator> allocator_;
+    std::unique_ptr<SafeAllocator> allocator_;
```

Comment:
HNSW object will hold the allocator instance now?

Reply:
use std::move to hold